QModManager/API/Messages.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace QModManager.API
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class MessageReceiver : Attribute
+    {
+        public QMod Sender;
+
+        public MessageReceiver(QMod sender)
+        {
+            Sender = sender ?? throw new ArgumentNullException("The provided mod is null!", nameof(sender));
+        }
+        public MessageReceiver(Assembly senderAssembly)
+        {
+            Sender = QModAPI.GetMod(senderAssembly, true, true) ?? throw new ArgumentException("The provided assembly is not a mod assembly!", nameof(senderAssembly));
+        }
+        public MessageReceiver(string senderID)
+        {
+            // Should we not want to throw an exception in case the mod is not present?
+            Sender = QModAPI.GetMod(senderID, true, true) ?? throw new ArgumentException($"No mod matching the provided ID \"{senderID}\" was found!", nameof(senderID));
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class GlobalMessageReciever : Attribute
+    {
+        public GlobalMessageReciever() { }
+    }
+
+    public static partial class QModAPI
+    {
+        public static void SendMessage(QMod mod, string message, params object[] data)
+        {
+            if (mod == null) return;
+            if (data == null) data = new object[] { };
+
+            QMod caller = GetMod(Assembly.GetCallingAssembly(), true);
+
+            if (mod.MessageReceivers.TryGetValue(caller, out List<MethodInfo> methods))
```

Comment:
Is the `MessageReceivers` field in the `QMod` class  guaranteed to be not-null by this point?  
Same concern on `BroadcastMessage` method here.

Reply:
No, it could actually be null... A solution would be to either discard messages that were sent before the mod was loaded or cache them and send them once it has loaded

Reply:
Which one of these do you think would be the way to go?

Reply:
With commit 83306b0, I implemented a check for null `MessageReceivers` in the `QMod` class.